site/docs/spec.md

```diff
@@ -548,6 +548,21 @@ Notes:
 1. An alternative, *strict projection*, creates a partition predicate that will match a file if all of the rows in the file must match the scan predicate. These projections are used to calculate the residual predicates for each file in a scan.
 2. For example, if `file_a` has rows with `id` between 1 and 10 and a delete file contains rows with `id` between 1 and 4, a scan for `id = 9` may ignore the delete file because none of the deletes can match a row that will be selected.
 
+#### Snapshot Reference
+
+Snapshot reference allows users to perform tagging and branching within an Iceberg table. The detailed user experience is described in [Iceberg Tagging and Branching](../snapshot-tag-branch).
+
+The snapshot reference object records all the user-defined information of a snapshot including name, reference type and retention policy configurations.
+
+| v2         | Field name                   | Type      | Description |
+| ---------- |------------------------------|-----------|-------------|
+| _required_ | **`snapshot-id`**            | `long`    | The ID of the snapshot referenced |
+| _required_ | **`name`**                   | `string`  | The name of the reference |
+| _required_ | **`type`**                   | `string`  | Type of the reference, `tag` or `branch` |
+| _optional_ | **`min-snapshots-to-keep`**  | `int`     | For `branch` type only, the minimum number of snapshots to keep in a branch |
```

Comment:
Will it make sense to mention the default, since it is an optional field.

Reply:
good point, added